### DISCUSSION TITLE
EcoWitt

### DISCUSSION THREAD

**Discussion by litojero** [breezy-weather/breezy-weather#1524] [Ideas]
### Describe your idea

Hello, it could be added to put data on our meteorological station, EcoWitt API type

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.3.1 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
Please explain how to access the API documentation, this website looks completely broken: https://api.ecowitt.net/

**Comment by litojero** [comment]
Hi Julien , look on this page to see if it works for you, Ecowitt is a
manufacturer of weather stations all over the world
https://doc.ecowitt.net/web/#/apiv3cn?page_id=2

**Reply by papjul** [reply]
From what I can see on the English website: https://doc.ecowitt.net/web/#/apiv3en?page_id=17

The URL is constructed that way:
`https://api.ecowitt.net/api/v3/device/real_time?application_key=APPLICATION_KEY&api_key=API_KEY&mac=YOUR_MAC_CODE_OF_DEVICE&call_back=all`

From what I understand, you can get `APPLICATION_KEY` and `API_KEY` by registering on this website: https://www.ecowitt.net/register

Please confirm this registration doesn't require a phone number or credit card info.

As for `YOUR_MAC_CODE_OF_DEVICE`, please clarify how we can get the list of stations and their coordinates.

**Comment by litojero** [comment]
Exactly each station owner gets his API and API Application of your
station, there is neither cards nor telephone.Thank you very much Julien
por tu interés
Lito

**Reply by papjul** [reply]
You said there were stations all over the world, please tell us where to find this list.
We need for each station:
- Mac code
- Latitude
- Longitude

Without this list, it can't be implemented.

**Comment by litojero** [comment]
If you want to see the data of the stations you register in Ecowitt which
is free and there are all the stations, for example in Spain.

**Reply by papjul** [reply]
I just registered. I indeed have access to a weather map with all public stations and their values.

Unfortunately, I didn't find how to access the `APPLICATION_KEY` and `API_KEY` on the website once you are logged in, could you help me?

Once done, we will have to confirm the following endpoint indeed lists all public stations, and not just the devices you own: https://doc.ecowitt.net/web/#/apiv3en?page_id=21

**Comment by litojero** [comment]
Hi Julien
When I open the Ecowitt website in Profile I can get the API and API Key

**Reply by papjul** [reply]
Thanks, I didn't think of clicking on the avatar.

Unfortunately, the endpoint I mentioned previously doesn't list any station, and there doesn't seem to be any other endpoint that would give us the list of all stations (or alternatively, the nearest station given a latitude and longitude).

Let us know if you find anything

**Comment by litojero** [comment]
Julien I think it works so that you have that station, enter your
credentials and see your station in your application.An example the
SmartMixin application integrates several, I hope I have explained better

**Comment by litojero** [comment]
https://play.google.com/store/apps/details?id=com.vtoupet.smartmixin

This is application but paid, but to give an example

**Reply by papjul** [reply]
SmartMixin is a paid application but they are able to access all public EcoWitt stations in the world for free, did I understand correctly? If so, how are they doing it? I need the API documentation

**Comment by litojero** [comment]
Julien In the documentation of the Ecowitt API is not how to implement it
in the application?The same has been complicated to insert EcoWitt but
there is also Wunderground that supports many station models

**Reply by papjul** [reply]
There is no endpoint for getting all public station list in the documentation.

Wunderground is only available to station owners or paid users, so it was rejected: see #620

**Comment by litojero** [comment]
Hello Julien, I can help you little more, it was an idea to integrate
weather stations around the world, thank you very much for your attention

**Reply by papjul** [reply]
Closing as not possible?

**Comment by litojero** [comment]
Julien I am not a developer but what you need I am to help you, test.

**Reply by papjul** [reply]
We can take care of the development, but we need the documentation of how to access the list of all public stations with the API.

**Comment by litojero** [comment]
Hi Julien, I sent an email to Ecowitt to tell him what you told me, I send
you a link from. Another brand of weather stations called Netatmo, look to
see what you think. Cheers

**Comment by litojero** [comment]
https://dev.netatmo.com/apidocumentation/weather

**Reply by papjul** [reply]
Please use a different thread for a different source.

Please also explain a bit the context in which you need Netatmo personal stations. While some may be good, most of them are probably not as good as professional ones, which must meet some criteria such as having no obstacle X meters around, shadow, etc. It may be fun to know what's the weather like in your backyard, but maybe not as useful to know someone else's backyard weather.

AEMET has an extensive network of professional stations in Spain with a better coverage than Netatmo, that is probably much better to use.

**Comment by litojero** [comment]
It was just that you looked at another option, my station is Ecowitt, when
they answered the mail I already told you
